### ISSUE TITLE
504 timeout when uploading custom certificates

### ISSUE THREAD

**Issue by SGStino** [jc21/nginx-proxy-manager#28]
I'm getting a classic nginx 504 Gateway Time-out error page response when adding an new custom certificate:

`POST http://proxy.docker/api/nginx/certificates/validate`

there are no log entries appearing during the period between submit request and timout response. However, there are 2 lines that don't look healthy:
```
[11/19/2018] [3:44:21 PM] [Access  ] › ✖  error                        users:get 1 Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
[11/19/2018] [3:44:21 PM] [Express ] › ⚠  warning                      Permission Denied
```
this also means that it's probably not issue #9 that returned?

Is there a place where I can discover more logging?

**Comment by SGStino** [comment]
Ok, Knex seems to be the connection to MariaDb?

So i looked into the maria logging and found this:
```2018-11-19 15:59:13 92 [Warning] Aborted connection 92 to db: 'npm' user: 'root' host: '172.17.0.1' (Got timeout reading communication packets) ```

a lot of them ..

why would it abort connections while also allowing others (since nginx proxy manager seems to work for some (most) tasks)?

**Comment by SGStino** [comment]
I've created mariadb with portainer. But if i were to guess from the settings it would be
`docker run -p 3306:3306 -v mariadb:/var/lib/mysql  mariadb:latest`

for "npm":
`docker run -p 81:81 -p 80:80 -p 443:443 -v /etc/letsencrypt:/etc/letsencrypt -v /etc/nginx-proxy-manager/config.json:/app/config/production.json -v /var/nginx-proxy-manager/data:/data jc21/nginx-proxy-manager:latest`

it's running on Ubuntu 18.04 on a VMWare ESX infra, so I'd guess it's not going to be hardware constraints from looking at the usage in vSphere. But there was a short period where the host ran out of diskspace for a while.

Does mariadb not recover from errors like that perhaps?


currently npm's output  contains the following: 
```
[11/28/2018] [2:47:01 PM] [Access  ] › ✖  error                        users:get 1 select `user`.* from `user` where `id` = 1 and `is_deleted` = 0 and `is_disabled` = 0 - Packets out of order. Got: 0 Expected: 1
[11/28/2018] [2:47:01 PM] [Express ] › ⚠  warning                      Permission Denied 
``` 

so a guess from someone not having any experience with mariadb:

did the temporary "out of diskspace" issue cause mariadb to generate bad records, that it's now blindly replaying to it's clients, who do check if the record packets are in order?

But I'm not entirely sure if those errors are even related to the certificates, because new attempts at uploading certificates don't generate any more log entries. Hence why I asked if I could find some more detailed logging somewhere? :) It looks more like the error is from the moment I tried to login (which did work).
